libdnf/hy-subject.cpp

```diff
@@ -349,13 +360,17 @@ hy_subject_get_best_solution(HySubject subject, DnfSack *sack, HyForm *forms, Hy
 
 HySelector
 hy_subject_get_best_sltr(HySubject subject, DnfSack *sack, HyForm *forms, bool obsoletes,
-                         const char *reponame)
+                         const char *reponame,
+                         bool with_src)
 {
     HyNevra nevra = NULL;
     HyQuery query = hy_subject_get_best_solution(subject, sack, forms, &nevra, FALSE, TRUE, TRUE,
-                                                 TRUE);
+                                                 TRUE, with_src);
     if (hy_query_is_not_empty(query)) {
-        hy_query_filter(query, HY_PKG_ARCH, HY_NEQ, "src");
+        // This inversion is because in the !with_src case, we already
+        // filtered above.
```

Comment:
I'm still confused; but then we're always filtering out src pkgs even if `with_src` is `TRUE`, right?